zerver/views/message_edit.py

```diff
@@ -143,13 +143,14 @@ def validate_can_delete_message(user_profile: UserProfile, message: Message) ->
     return
 
 
+@transaction.atomic
 @has_request_variables
 def delete_message_backend(
```

Comment:
This isn't the only relevant codepath - e.g. there's also the "delete topic" API where I'm pretty sure we'll want to lock too. `git grep do_delete_messages` should give all the places we need to take a look at.

Reply:
Updated to handle the `delete topic` codepath too.